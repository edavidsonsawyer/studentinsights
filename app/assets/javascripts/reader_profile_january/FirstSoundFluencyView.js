import React from 'react';
import expandedViewPropTypes from './expandedViewPropTypes';

export default class FirstSoundFluencyView extends React.Component {
  render() {
    const {student} = this.props;

    const materialsEl = (
      <img
        width="100%"
        style={{border: '1px solid #ccc'}}
        src="/assets/reading/FirstSoundFluency-k.jpg"
      />
    );
    return (
      <div style={styles.root}>
        <div style={styles.header}>
          <div style={styles.title}>First Sound Fluency</div>
          <div style={styles.close}>Close</div>
        </div>
        <div style={styles.columns}>
          <div style={{...styles.column, ...styles.materials}}>
            <div>{materialsEl}</div>
          </div>
          <div style={{...styles.column, ...styles.strategies}}>
            <div style={styles.subtitle}>Instructional strategies</div>
            <div>...strategies...</div>
          </div>
          <div style={{...styles.column, ...styles.data}}>
            <div style={styles.subtitle}>Data for {student.first_name}</div>
            <div>...data...</div>
          </div>
        </div>
      </div>
    );
  }
}
FirstSoundFluencyView.propTypes = expandedViewPropTypes;

const styles = {
  root: {
    padding: 10
  },
  header: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingBottom: 10
  },
  title: {
    fontWeight: 'bold',
    fontSize: 16
  },
  close: {
    color: '#999'
  },
  columns: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'space-between'
  },
  column: {
    marginRight: 20
  },
  materials: {
    flex: 5
  },
  strategies: {
    flex: 3
  },
  data: {
    flex: 5
  },
  subtitle: {
    color: '#333',
    marginBottom: 10
  }
};