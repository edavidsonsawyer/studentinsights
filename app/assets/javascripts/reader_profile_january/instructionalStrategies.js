import _ from 'lodash';

// Keys for describing the reading categories.
export const STUDENT_EXPERIENCE = 'c:STUDENT_EXPERIENCE';
export const ORAL_LANGUAGE = 'c:ORAL_LANGUAGE';
export const PHONOLOGICAL_AWARENESS = 'c:PHONOLOGICAL_AWARENESS';
export const PHONICS_FLUENCY = 'c:PHONICS_FLUENCY';
export const COMPREHENSION = 'c:COMPREHENSION';


// Map the downcased text that people use in Excel to internal keys,
// so code outside this file can only use keys.
const CategoryKeyMapping = {
  'phonological awareness': PHONOLOGICAL_AWARENESS,
  'phonics fluency': PHONICS_FLUENCY
};

// Takes the CSV format, but parses the grades field into an array
// that uses Student Insights-style grade values (eg, KF for kindergarten).
let instructionalStrategies = null;
export function readInstructionalStrategies() {
  if (instructionalStrategies === null) {
    instructionalStrategies = _.flatMap(RAW_RECORDS, row => {
      const categoryKey = CategoryKeyMapping[row.category_text.toLowerCase()];
      if (!categoryKey) return [];
      return [{
        ...row,
        category_key: categoryKey,
        grades: row.grades.split(' ').map(g => g === 'K' ? 'KF' : g)
      }];
    });
  }

  return instructionalStrategies;
}

// Instructional strategies are by category and grade, generically.
// Also filter ones that don't have text or educator_email.
export function matchStrategies(strategies, grade, categoryKey) {
  return strategies.filter(strategy => {
    if (strategy.grades.indexOf(grade) === -1) return false;
    if (strategy.category_key !== categoryKey) return false;
    if (strategy.text === '') return false;
    if (strategy.educator_email === '') return false;

    return true;
  });
}


/*
const fs = require('fs');
const csv = fs.readFileSync('strategies.csv').toString();
const parse = require('csv-parse/lib/sync');
const records = parse(csv, {
  columns: true,
  skip_empty_lines: true
});
console.log(JSON.stringify(records.slice(2), null, 2));
*/
const RAW_RECORDS = [
  {
    "category_text": "Phonological Awareness",
    "grades": "K 1 2 3 4 5",
    "text": "SPS Heggerty",
    "description": "5 minute lessons for phonological awareness",
    "educator_email": "ggeorge",
    "url": "https://www.dropbox.com/sh/md15rd36xq81wwi/AAD-I9Eto1rG5wdgwN3bxOEwa/ELA%202019-2020/Kindergarten/PA%20Intervention"
  },
  {
    "category_text": "Phonological Awareness",
    "grades": "K",
    "text": "SPS notebook for double dose",
    "description": "Notebook for repeating Kindergarten PA unit, 12 weeks",
    "educator_email": "jbuckwalter",
    "url": "https://www.dropbox.com/sh/md15rd36xq81wwi/AAD40J3eTcxj5khUtz5mpRKva/ELA%202019-2020/Kindergarten/PA%20Workbooks"
  },
  {
    "category_text": "Phonological Awareness",
    "grades": "1 2 3",
    "text": "LIPS",
    "description": "Intensive 3x/week program focused on sound production",
    "educator_email": "uharel",
    "url": "https://lindamoodbell.com/program/lindamood-phoneme-sequencing-program"
  },
  {
    "category_text": "Phonics Fluency",
    "grades": "K 1",
    "text": "SPS Lively Letters",
    "description": "Cards for practicing letter naming",
    "educator_email": "cconnolly",
    "url": "https://www.dropbox.com/preview/ela%20folder/Lively%20Letters%20SPS%20version/SPS%20Lively%20Letters.pdf?role=personal"
  },
  {
    "category_text": "Phonics Fluency",
    "grades": "K 1 2 3",
    "text": "CVC Blending notebook",
    "description": "Notebook for practicing CVC blends",
    "educator_email": "mjouvelakas",
    "url": "https://drive.google.com/drive/folders/1E9ZNh3gJzRWZwijF-VYI-In3M8dEAjJa"
  },
  {
    "category_text": "Phonics Fluency",
    "grades": "1 2 3",
    "text": "Snap Words",
    "description": "5min lessons with visual stories for learning sight words",
    "educator_email": "jbuckwalter",
    "url": "https://drive.google.com/open?id=1SjAypbcoTyUVQ6wCsKSYVGnEvlCSLicu"
  },
  {
    "category_text": "Phonics Fluency",
    "grades": "4 5",
    "text": "Rewards",
    "description": "25 lesson, multi-syllabic decoding fluency intervention",
    "educator_email": "uharel",
    "url": "https://www.voyagersopris.com/literacy/rewards/overview"
  },
  {
    "category_text": "Phonics Fluency",
    "grades": "2 3 4 5",
    "text": "Inside Phonics",
    "description": "For ELL students, daily systematic intervention",
    "educator_email": "pmitropoulos",
    "url": "https://ngl.cengage.com/search/productOverview.do?N=4294918395+201+4294918868&Ns=P_Product_Title%7C0&Ntk=P_EPI&Ntt=16168729182104448189494671682736665835&Ntx=mode%2Bmatchallpartial&homePage=false"
  },
  {
    "category_text": "Oral language",
    "grades": "2 3 4 5",
    "text": "Visualizing Verbalizing",
    "description": "Listening comprehension and oral language program",
    "educator_email": "uharel",
    "url": "https://lindamoodbell.com/program/visualizing-and-verbalizing-program"
  }
];
