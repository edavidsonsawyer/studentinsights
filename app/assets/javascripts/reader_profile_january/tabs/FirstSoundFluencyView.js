import React from 'react';
import expandedViewPropTypes from './expandedViewPropTypes';

export default class FirstSoundFluencyView extends React.Component {
  render() {
    return 'moar first sound fluency';
  }
}
FirstSoundFluencyView.propTypes = expandedViewPropTypes;
