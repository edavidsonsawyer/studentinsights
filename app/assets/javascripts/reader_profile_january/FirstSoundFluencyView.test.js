import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {withNowContext} from '../testing/NowContainer';
import PerDistrictContainer from '../components/PerDistrictContainer';
import FirstSoundFluencyView from './FirstSoundFluencyView';
import {readInstructionalStrategies} from './instructionalStrategies';


// in spring of KF, looking back at fall and winter scores
const TEST_NOW_STRING = '2018-05-15T11:03:06.123Z';

export function testProps(props = {}) {
  return {
    onClose: jest.fn(),
    student: {
      id: 123,
      first_name: 'Max',
      grade: 'KF'
    },
    readerJson: {
      access: {},
      services: [],
      iep_contents: null,
      feed_cards: [],
      current_school_year: 2017,
      benchmark_data_points: [{
        "id": 1001,
        "student_id": 123,
        "benchmark_school_year": 2017,
        "benchmark_period_key": 'fall',
        "benchmark_assessment_key": "dibels_fsf",
        "json": {
          "value": 12
        },
        "educator_id": 999999,
        "created_at": "2017-09-22T19:27:10.429Z",
        "updated_at": "2017-09-22T19:27:10.429Z"
      }, {
        "id": 1002,
        "student_id": 123,
        "benchmark_school_year": 2017,
        "benchmark_period_key": 'winter',
        "benchmark_assessment_key": "dibels_fsf",
        "json": {
          "value": 36
        },
        "educator_id": 999999,
        "created_at": "2018-01-25T19:27:10.429Z",
        "updated_at": "2018-01-25T19:27:10.429Z"
      }]
    },
    instructionalStrategies: readInstructionalStrategies(),
    ...props
  };
}

export function testRender(props = {}) {
  return (
    withNowContext(TEST_NOW_STRING, (
      <PerDistrictContainer districtKey="somerville">
        <FirstSoundFluencyView {...props} />
      </PerDistrictContainer>
    ))
  );
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  ReactDOM.render(testRender(testProps()), el);
});


it('snapshots', () => {
  const tree = renderer.create(testRender(testProps())).toJSON();
  expect(tree).toMatchSnapshot();
});
