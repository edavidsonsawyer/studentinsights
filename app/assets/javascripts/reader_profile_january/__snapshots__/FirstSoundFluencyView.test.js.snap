// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  className="ExpandedLayout"
  style={
    Object {
      "padding": 10,
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      First sound fluency
    </div>
    <div
      onClick={[MockFunction]}
      style={
        Object {
          "color": "#999",
          "cursor": "pointer",
        }
      }
    >
      Close
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 5,
          "marginRight": 20,
        }
      }
    >
      <div>
        <div
          className="MaterialsCarousel"
        >
          <img
            className="MaterialImage"
            src="/img/reading/FirstSoundFluency-KF-winter.jpg"
            style={
              Object {
                "border": "1px solid #ccc",
              }
            }
            width="100%"
          />
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "flex": 3,
          "marginRight": 20,
          "overflow": "hidden",
        }
      }
    >
      <div
        style={
          Object {
            "color": "#333",
            "marginBottom": 10,
          }
        }
      >
        Instructional strategies
      </div>
      <div>
        <div
          className="Strategies"
        >
          <div>
            <a
              href="https://www.dropbox.com/sh/md15rd36xq81wwi/AAD-I9Eto1rG5wdgwN3bxOEwa/ELA%202019-2020/Kindergarten/PA%20Intervention"
              rel="noopener noreferrer"
              style={
                Object {
                  "alignItems": "flex-start",
                  "background": "#eee",
                  "border": "1px solid #ccc",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": "4em",
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "padding": 10,
                }
              }
              target="_blank"
              title="5 minute lessons for phonological awareness"
            >
              <div
                style={
                  Object {
                    "fontWeight": "bold",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                SPS Heggerty
              </div>
              <div>
                by 
                ggeorge
              </div>
            </a>
            <a
              href="https://www.dropbox.com/sh/md15rd36xq81wwi/AAD40J3eTcxj5khUtz5mpRKva/ELA%202019-2020/Kindergarten/PA%20Workbooks"
              rel="noopener noreferrer"
              style={
                Object {
                  "alignItems": "flex-start",
                  "background": "#eee",
                  "border": "1px solid #ccc",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": "4em",
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "padding": 10,
                }
              }
              target="_blank"
              title="Notebook for repeating Kindergarten PA unit, 12 weeks"
            >
              <div
                style={
                  Object {
                    "fontWeight": "bold",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                SPS notebook for double dose
              </div>
              <div>
                by 
                jbuckwalter
              </div>
            </a>
          </div>
          <div
            style={
              Object {
                "color": "#999",
                "fontSize": 12,
                "margin": 10,
              }
            }
          >
            Add suggestion
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "flex": 5,
          "marginRight": 20,
        }
      }
    >
      <div
        style={
          Object {
            "color": "#333",
            "marginBottom": 10,
          }
        }
      >
        Data for 
        Max
      </div>
      <div>
        <div>
          <div>
            <div>
              <b
                style={
                  Object {
                    "paddingRight": 5,
                  }
                }
              >
                36
              </b>
              <span>
                in 
                winter
                 of 
                Kindergarten
              </span>
            </div>
          </div>
          <div>
            <div>
              <b
                style={
                  Object {
                    "paddingRight": 5,
                  }
                }
              >
                12
              </b>
              <span>
                in 
                fall
                 of 
                Kindergarten
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
